### PR TITLE
Replace os() with canImport(Darwin)

### PR DESCRIPTION
**Pull request by brennobemoura**
This will enable compiling the package to `xrOS` using the right code.

### CODE REVIEW

**Review by FranzBusch — APPROVED** [review]

**Comment by Lukasa** [comment]
@swift-server-bot add to allowlist

**Comment by FranzBusch** [comment]
@swift-server-bot test this please

**Comment by brennobemoura** [comment]
@FranzBusch is this test failure related to the modification I've made?

**Comment by FranzBusch** [comment]
No this looks like our allocation counters need to be updated because we regressed somewhere. We should check where that allocation regression comes from.

File: Sources/NIOHPACK/HuffmanCoding.swift [hunk 1]
```diff
@@ -268,7 +268,7 @@ extension String {
 // this declaration on having the 5.3 compiler. This has caused a number of build issues. While updating
 // to newer Xcodes does work, we can save ourselves some hassle and just wait until 5.4 to get this
 // enhancement on Apple platforms.
-#if (compiler(>=5.3) && !(os(macOS) || os(iOS) || os(tvOS) || os(watchOS))) || compiler(>=5.4)
+#if (compiler(>=5.3) && !(canImport(Darwin))) || compiler(>=5.4)
```
Comment (Lukasa) [inline Sources/NIOHPACK/HuffmanCoding.swift:271]: We can drop this condition entirely now, as we no longer support Swift versions older than 5.6, so we will always enter this branch.

**Review by Lukasa — CHANGES_REQUESTED** [review]

**Comment by Lukasa** [comment]
Alloc limits fix in #400

File: Sources/NIOHPACK/HuffmanCoding.swift (hunk 1 above)
Comment (brennobemoura) [reply Sources/NIOHPACK/HuffmanCoding.swift:271]: Fixing this right now.

**Comment by brennobemoura** [comment]
@Lukasa great catch! You nailed it.

**Review by Lukasa — APPROVED** [review]
Love it, negative diffs are the best. :sparkles:

**Pull request merged**